src/clashing_inputs.rs

```diff
@@ -47,36 +46,87 @@ impl ClashStrategy {
     }
 }
 
-impl UserInput {
-    /// Does `self` clash with `other`?
-    #[must_use]
-    fn clashes(&self, other: &UserInput) -> bool {
-        use UserInput::*;
+/// The basic inputs that make up a [`UserInput`].
+#[derive(Debug, Clone)]
+#[must_use]
+pub enum BasicInputs {
+    /// The input consists of a single, fundamental [`UserInput`].
+    /// In most cases, the input simply holds itself.
+    Simple(Box<dyn UserInput>),
+
+    /// The input consists of multiple independent [`UserInput`]s.
```

Comment:
I don't understand the distinction between Composite and Group.

Reply:
This is about handling input conflicts. If I understand correctly, when Input A and Input B conflict, one of them is a true subset of the other.

`Group` is used for Chords and other similar input combinations (like `Sequences` in future), and these are made up of one or more inputs.
`Composite` is for inputs that are composed of at least two inputs, like `VirtualAxis` and `VirtualDPad`, which are made up of two and four buttons, respectively.

But maybe it could be changed to return a Vec and just check the lengths and basic inputs of each.

Reply:
Ah, got it.